Gems/GradientSignal/Code/Tests/GradientSignalTestMocks.h

```diff
@@ -201,4 +205,52 @@ namespace UnitTest
         SurfaceData::SurfaceDataRegistryHandle m_providerHandle = SurfaceData::InvalidSurfaceDataRegistryHandle;
     };
 
+
+    // The following Mock class of GradientSignal::PerlinImprovedNoise eliminates the random shuffling of the initial permutation
+    // table since the randomness function (std::mt19937) is only deterministic per platform or std c++ library implementation.
+    // By taking out the randomness, this makes the perlin test deterministic across all platforms
+    class MockPerlinImprovedNoise : public GradientSignal::PerlinImprovedNoise
+    {
+    public:
+        MockPerlinImprovedNoise(int seed) :
+            GradientSignal::PerlinImprovedNoise(seed)
+        {
+            PrepareTable(seed);
+        }
+
+        virtual ~MockPerlinImprovedNoise() = default;
+
+    protected:
+
+        void PrepareTable([[maybe_unused]] int seed) override
+        {
+            AZStd::array<int, 256> randtable;
+            std::iota(randtable.begin(), randtable.end(), 0);
+            // Normally the following random shuffle occurs, but suppress it for unit testing
+            // std::shuffle(randtable.begin(), randtable.end(), std::mt19937(seed));
+            for (int x = 0; x < 256; ++x)
+            {
+                m_permutationTable[x] = randtable[x];
+                m_permutationTable[x + 256] = randtable[x];
```

Comment:
Unclear why it's important to have this mock set up random values. (at least without digging into the parent class to understand what this is bypassing/clobbering) What is the benefit of random values over simply providing static values?

Reply:
Personally I would rather refactor this class to make it unit testable. And also the original test uses known values as to what it expects as a valid random seed, but doesn’t seem to me like a real legitimate test against the algorithm. That said, the purpose of this PR was to not alter the original code as that would impose risk to production code which is not what I want to do. I cannot answer the reason for the random shuffling if that is even part of the perlin algorithm. The reason to go down the route of eliminating the random shuffle was because that turned out to be platform and distro specific. The other option is to disable the test

Reply:
Okay, after looking at the Wikipedia definition of [perlin](https://en.m.wikipedia.org/wiki/Perlin_noise) the randomness is part of the input to the perlin algorithm. So maybe we can refactor to use a random table that is passed into the component and then processed. That way a known test table of numbers can be used for every platform.

Reply:
Those are not random values. The randomization is done with ```std::shuffle```, not ```std::iota```. This produces a fixed array of increasing digits. Maybe it will make more sense if this was changed to used a fixed static array of digits instead of relying on std::iota to produce it

Reply:
Another possible approach is to refactor the code and break out the ```PerlinImprovedNoise``` to instead of taking in a seed, taking in a POD that represents the matrix of numbers that will be used in the perlin processing. The production version of the component will then generate this POD via a seed + the random shuffling, and the unit test will use a mock version that is a fixed value instead